test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ComplexTypeModelBinderTest.cs

```diff
@@ -350,6 +350,52 @@ public void CreateModel_InstantiatesInstanceOfMetadataType()
         }
 
         [Fact]
+        public void CreateModel_ForStructModelType_AsTopLevelObject_ThrowsException()
```

Comment:
We already have more tests for these here https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.IntegrationTests/ActionParametersIntegrationTest.cs#L387-L461